File: leetcode/solutions/0733.FloodFill/floodFill_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import floodFill from "./floodFill.ts";

Deno.test("0733. Flood Fill", () => {
  type args = [number[][], number, number, number];
  type test = [args, number[][]];
  const tests: test[] = [
    [
      [
        [
          [1, 1, 1],
          [1, 1, 0],
          [1, 0, 1],
        ],
        1,
        1,
        2,
      ],
      [
        [2, 2, 2],
        [2, 2, 0],
        [2, 0, 1],
      ],
    ],
    [
      [
        [
          [0, 0, 0],
          [0, 0, 0],
        ],
        0,
        0,
        2,
      ],
      [
        [2, 2, 2],
        [2, 2, 2],
      ],
    ],
    [
      [
        [
          [0, 0, 0],
          [0, 1, 1],
        ],
        1,
        1,
        1,
      ],
      [
        [0, 0, 0],
        [0, 1, 1],
      ],
    ],
  ];

  for (const [args, want] of tests) {
    assertEquals(floodFill(...args), want);
  }
});

File: leetcode/solutions/0733.FloodFill/floodFill_test.go
package solution

import (
	"testing"

	"github.com/google/go-cmp/cmp"
)

func Test_0733_FloodFill(t *testing.T) {
	t.Parallel()
	type in struct {
		arg1 [][]int
		arg2 int
		arg3 int
		arg4 int
	}
	tests := []struct {
		in  in
		out [][]int
	}{
		{
			in{
				[][]int{
					{1, 1, 1},
					{1, 1, 0},
					{1, 0, 1},
				},
				1,
				1,
				2,
			},
			[][]int{
				{2, 2, 2},
				{2, 2, 0},
				{2, 0, 1},
			},
		},
		{
			in{
				[][]int{
					{0, 0, 0},
					{0, 0, 0},
				},
				0,
				0,
				2,
			},
			[][]int{
				{2, 2, 2},
				{2, 2, 2},
			},
		},
		{
			in{
				[][]int{
					{0, 0, 0},
					{0, 1, 1},
				},
				1,
				1,
				1,
			},
			[][]int{
				{0, 0, 0},
				{0, 1, 1},
			},
		},
	}

	for _, tt := range tests {
		tt := tt

		t.Run("Test", func(t *testing.T) {
			t.Parallel()

			got := floodFill(tt.in.arg1, tt.in.arg2, tt.in.arg3, tt.in.arg4)

			if !cmp.Equal(got, tt.out) {
				t.Errorf("got %#v want %#v", got, tt.out)
			}
		})
	}
}
